src/app/core/services/shopping-cart-service/shopping-cart-service.service.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ShoppingCartServiceService {
+
+  private _selectedProductsList: BehaviorSubject<Array<number>> = new BehaviorSubject<Array<number>>([]);
```

Comment:
Better like this:
private _selectedProductsList: BehaviorSubject<number[]> = new BehaviorSubject([]);